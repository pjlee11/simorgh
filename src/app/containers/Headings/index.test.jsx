import React from 'react';
import HeadingsContainer from './index';
import { textBlock } from '../../models/blocks';
import {
  shouldShallowMatchSnapshot,
  isNull,
} from '../../helpers/tests/testHelpers';

const template = (title, text, type) => {
  describe(title, () => {
    const data = {
      ...textBlock(text),
      type,
    };
    shouldShallowMatchSnapshot(
      'should render correctly',
      <HeadingsContainer {...data} />,
    );
  });
};

describe('Headings', () => {
  describe('with no data', () => {
    isNull('should not render anything', <HeadingsContainer />);
  });

  template('with headline data', 'This is a headline!', 'headline');
  template('with subheadline data', 'This is a subheadline', 'subheadline');
});
