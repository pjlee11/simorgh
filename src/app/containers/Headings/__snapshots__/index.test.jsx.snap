// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headings with headline data should render correctly 1`] = `
<Headline
  text="This is a headline!"
>
  This is a headline!
</Headline>
`;

exports[`Headings with subheadline data should render correctly 1`] = `
<SubHeading
  text="This is a subheadline"
>
  This is a subheadline
</SubHeading>
`;
